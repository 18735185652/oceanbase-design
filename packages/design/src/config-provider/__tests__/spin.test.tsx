import React from 'react';
import { render } from '@testing-library/react';
import ConfigProvider from '..';
import Spin from '../../spin';

describe('ConfigProvider spin', () => {
  beforeEach(() => {});

  it('ConfigProvider spin.indicator should work', () => {
    const { container, asFragment } = render(
      <ConfigProvider
        spin={{
          indicator: <div className="custom-indicator" />,
        }}
      >
        <Spin />
      </ConfigProvider>
    );
    expect(container.querySelector<HTMLElement>('.ant-spin-dot')?.className).toContain(
      'custom-indicator'
    );
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
