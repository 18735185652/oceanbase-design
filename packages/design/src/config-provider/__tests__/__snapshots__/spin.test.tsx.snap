// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigProvider spin spin.indicator should work 1`] = `
<div
  class="ant-app"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1w4v3hc"
  >
    <div
      class="custom-indicator ant-spin-dot"
    />
  </div>
</div>
`;
